subprojects/build-operations/src/main/java/org/gradle/trace/buildops/BuildOperationTrace.java

```diff
@@ -46,6 +57,51 @@ public void started(BuildOperationDescriptor buildOperationDescriptor, Operation
                 writer.finished();
             }
         }
+    }
+
+    private static class BuildOperationDurationRecordingListener extends DetachingBuildOperationListener {
+
+        private final Class<?> capturedBuildOperation;
+
+        private AsyncWriter<Long> writer;
+
+        BuildOperationDurationRecordingListener(String capturedBuildOperation, File outFile, BuildOperationListenerManager manager) throws ClassNotFoundException {
+            super(outFile, manager);
+            this.capturedBuildOperation = Class.forName(capturedBuildOperation + "$Details");
+        }
+
+        @Override
+        public void finished(BuildOperationDescriptor buildOperationDescriptor, OperationFinishEvent operationFinishEvent) {
+            Object details = buildOperationDescriptor.getDetails();
+            if (buildOperationDescriptor.getDetails() != null && capturedBuildOperation.isAssignableFrom(details.getClass())) {
+                AsyncWriter<Long> writer = ensureWriter();
+                writer.append(operationFinishEvent.getEndTime() - operationFinishEvent.getStartTime());
```

Comment:
I'd expect us to collect the times in-memory and only write the end-result to a file. If we did that we could write the results to the same file in a map format instead of needed a separate file for each measured build operation.

What is the reason with writing each event to a file?

Reply:
I did it like this since it seemed more straight forward. I can also capture the (sum of) the measurements in memory and then write the result to a file.

If I want to write all results to the same file, then I need to either have
- one listener
- synchronize the listeners

I don't see much benefit in that. What would be the advantage of writing all the results into one file?

Reply:
You don't have to synchronize on the writes; you could simply collect the results in some concurrent structure and write them out at the end of the build instead.

I don't much care about the number of files we end up with (though a single text file would be more tidy and easier to process for a human). What I care about is that we should keep writing to files during the benchmark to a minimum.

Reply:
I am now only writing the cumulative time into the files. I think having one file per build operation we track is not too much overhead.